src/crimson/common/config_proxy.cc

```diff
@@ -4,13 +4,16 @@
 
 namespace ceph::common {
 
-ConfigProxy::ConfigProxy(const EntityName& name, std::string_view cluster)
+ConfigProxy::ConfigProxy(const EntityName& name, std::string_view cluster, ConfigValues* config_values)
 {
   if (seastar::engine().cpu_id() != 0) {
     return;
   }
-  // set the initial value on CPU#0
-  values.reset(seastar::make_lw_shared<ConfigValues>());
+  if (!config_values)
+    // set the initial value on CPU#0
+    values.reset(seastar::make_lw_shared<ConfigValues>());
+  else
+    values.reset(seastar::make_lw_shared<ConfigValues>(std::move(*config_values)));
```

Comment:
@tchaikov so what will be right here? *(std::move(config_values))?

Reply:
strictly speaking, it's difficult to get this right. unlike `std::shared_ptr<>`, `lw_shared_ptr<>` cannot be constructed from a raw pointer. so we cannot transfer the ownership from a `unique_ptr<>` to `lw_shared_ptr<>`. anyway, since we don't need to pre-populate `ConfigValues`, this is not a problem anymore.